src/open_producten/utils/tests/test_api_oidc.py

```diff
@@ -0,0 +1,136 @@
+from functools import partial
+from pathlib import Path
+
+from django.test import TestCase, override_settings
+from django.urls import reverse
+
+import requests
+import vcr
+from mozilla_django_oidc_db.models import OpenIDConnectConfig
+from rest_framework.exceptions import ErrorDetail
+from rest_framework.status import HTTP_200_OK, HTTP_401_UNAUTHORIZED
+
+from ...accounts.tests.factories import UserFactory
+from ...producttypen.tests.factories import ProductTypeFactory
+from .keycloak import mock_oidc_db_config
+
+TEST_FILES = (Path(__file__).parent / "vcr_cassettes").resolve()
+
+mock_admin_oidc_config = partial(
+    mock_oidc_db_config,
+    app_label="mozilla_django_oidc_db",
+    model="OpenIDConnectConfig",
+    id=1,
+    make_users_staff=True,
+    username_claim=["preferred_username"],
+)
+
+
+@override_settings()
+class TestApiOidcAuthentication(TestCase):
+
+    def setUp(self):
+        ProductTypeFactory.create()
+        UserFactory.create(superuser=True, username="testtest")
+
+    def generate_token_with_password(self, config):
+
+        payload = {
+            "client_id": config.oidc_rp_client_id,
+            "client_secret": config.oidc_rp_client_secret,
+            "username": "testtest",
+            "password": "test",
+            "grant_type": "password",
+        }
+
+        response = requests.post(
+            config.oidc_op_token_endpoint,
+            data=payload,
+        )
+
+        return response.json()["access_token"]
+
+    def generate_client_credentials(self, config):
+        payload = {
+            "client_id": config.oidc_rp_client_id,
+            "client_secret": config.oidc_rp_client_secret,
+            "grant_type": "client_credentials",
+        }
+
+        response = requests.post(
+            config.oidc_op_token_endpoint,
+            data=payload,
+        )
+
+        return response.json()["access_token"]
+
+    @vcr.use_cassette(str(TEST_FILES / "valid_token"))
```

Comment:
To avoid having to add this decorator for each test, you could maybe add a mixin similar to `OFVCRMixin` https://github.com/open-formulieren/open-forms/blob/4e6d5c17b14580ef4a7467906a7f38c19ce51f5a/src/openforms/contrib/brk/tests/test_client.py#L14-L15

Reply:
decided to keep the decorator for the mvp, may change it if more tests are added later.